scripts/enrich-junit-xml.sh

```
@@ -0,0 +1,285 @@
+# Write the CI properties to a temp file for injection into JUnit XML.
+# Uses a file-based approach instead of awk -v to avoid macOS awk issues
+# with multi-line strings.
+PROPS_FILE=$(mktemp)
+trap 'rm -f "$PROPS_FILE"' EXIT
+
+write_properties_file() {
+    local indent="$1"
+    local out="$2"
+
+    echo "${indent}<property name=\"ci.infra_provider\" value=\"${INFRA_PROVIDER}\"/>" > "$out"
+    echo "${indent}<property name=\"ci.deployment_env\" value=\"${DEPLOYMENT_ENV}\"/>" >> "$out"
```

Shellcheck (0.11.0)

[style] Consider using { cmd1; cmd2; } >> file instead of individual redirects.

(SC2129)